helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -90,61 +96,84 @@ private BestPossibleStateOutput compute(ClusterEvent event, Map<String, Resource
 
     BestPossibleStateOutput output = new BestPossibleStateOutput();
 
-    for (String resourceName : resourceMap.keySet()) {
-      logger.debug("Processing resource:" + resourceName);
+    // Reset current INIT/RUNNING tasks on participants for throttling
+    JobRebalancer.resetActiveTaskCount(cache.getLiveInstances().keySet(), currentStateOutput);
 
-      Resource resource = resourceMap.get(resourceName);
-      // Ideal state may be gone. In that case we need to get the state model name
-      // from the current state
-      IdealState idealState = cache.getIdealState(resourceName);
+    PriorityQueue<JobResourcePriority> jobResourceQueue = new PriorityQueue<JobResourcePriority>();
```

Comment:
We can use the queue for all resources, separating only task-related resources and puting it into the queue makes the logic less generic here.